db/version_set.cc

```diff
@@ -3992,6 +3996,8 @@ Status VersionSet::ProcessManifestWrites(
       if (!io_s.ok()) {
```

Comment:
I'm thinking we should keep the first IO or Status error occurred.
if (io_status_.ok()) {
  io_status_ = io_s;
} 
if (s.ok()) {
  s = io_s;
}

What do you think?

Reply:
Not sure which line this is for?

Reply:
L3981 and L3996. If we check io_status_ is not OK, it will not be updated by io_s. In this way, we keep the earliest error. So we do not need to check `if (io_status_.IsIOError())`

Reply:
In line 3987, we need to update `io_status_` because (1) `io_s.ok()`, (2) `io_status_.IsIOError()`. It means writing MANIFEST has recovered from a prior IOError by writing to a new MANIFEST successfully.

Reply:
I see. I got the logic here. Thanks for the explanation